Source/Scene/Scene.js

```diff
@@ -1698,6 +1716,95 @@ define([
         return object;
     };
 
+    var scratchNDC = new Cartesian4();
+    var scratchWorldCoords = new Cartesian4();
+
+    /**
+     * Returns the cartesian position reconstructed from the depth buffer and window position.
+     *
+     * @param {Cartesian2} windowPosition Window coordinates to perform picking on.
+     * @param {Cartesian3} [result] The object on which to restore the result.
+     * @returns {Cartesian3} The cartesian position.
+     *
+     * @exception {DeveloperError} windowPosition is undefined.
+     * @exception {DeveloperError} Picking from the depth buffer is not supported.
+     */
+    Scene.prototype.pickDepth = function(windowPosition, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if(!defined(windowPosition)) {
+            throw new DeveloperError('windowPosition is undefined.');
+        }
+        if (!this._globeDepth.supported) {
+            throw new DeveloperError('Picking from the depth buffer is not supported.');
+        }
+        //>>includeEnd('debug');
+
+        var context = this._context;
+        var uniformState = context.uniformState;
+
+        var drawingBufferPosition = SceneTransforms.transformWindowToDrawingBuffer(this, windowPosition, scratchPosition);
+        drawingBufferPosition.y = this.drawingBufferHeight - drawingBufferPosition.y;
+
+        var depth;
+        var index;
+        var numFrustums = this.numberOfFrustums;
+
+        for (var i = 0; i < numFrustums; ++i) {
+            var pickDepth = getPickDepth(this, context, i);
+            var pixels = context.readPixels({
+                x : drawingBufferPosition.x,
+                y : drawingBufferPosition.y,
+                width : 1,
+                height : 1,
+                framebuffer : pickDepth.framebuffer
+            });
+
+            depth = pixels[0] / 255.0 + pixels[1] / 65535.0 + pixels[2] / 16777215.0;
+            if (depth > 0.0 && depth < 1.0) {
+                index = i;
+                break;
+            }
+        }
+
+        if (!defined(index)) {
+            return undefined;
+        }
+
+        var camera = this._camera;
+
+        // Create a working frustum from the original camera frustum.
+        var frustum;
+        if (defined(camera.frustum.fov)) {
+            frustum = camera.frustum.clone(scratchPerspectiveFrustum);
+        } else if (defined(camera.frustum.infiniteProjectionMatrix)){
+            frustum = camera.frustum.clone(scratchPerspectiveOffCenterFrustum);
+        } else {
+            throw new DeveloperError('An orthographic projection matrix is not invertible.');
+        }
+
+        var renderedFrustum = this._frustumCommandsList[index];
+        frustum.near = renderedFrustum.near * 0.99;
+        frustum.far = renderedFrustum.far;
+        uniformState.updateFrustum(frustum);
+
+        var viewport = uniformState.viewport;
```

Comment:
Make this a separate function perhaps in `Transforms` since it goes well with [pointToWindowCoordinates](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Transforms.js#L740).  It can be `@private` until it is needed.